src/libs/tools/src/plugindatabase.cpp

```diff
@@ -379,6 +380,108 @@ std::vector<PluginSpec> ModulesPluginDatabase::lookupAllProvides (std::string co
 }
 
 
+class PluginVariantsDatabase::Impl
+{
+public:
+	Impl ()
+	{
+	}
+	~Impl ()
+	{
+	}
+	Modules modules;
```

Comment:
You do not have to inherit the Impl class? The `impl` needs a different name then..